charts/px-backup/templates/pxcentral-pre-install-hook.yaml

```diff
@@ -0,0 +1,113 @@
+{{/* Setting defaults if they are omitted. */}}
+{{- $customRegistryEnabled := .Values.images.customRegistryEnabled | default false }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pxcentral-onprem-job
+  namespace: {{ .Values.namespace }}
+  annotations:
+    "helm.sh/hook": "pre-install"
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+  labels:
+    app.kubernetes.io/component: pxcentral-pre-install-hook
+{{- include "px-backup.labels" . | nindent 4 }}
+---
+kind: ClusterRole
```

Comment:
Why does this need to be a cluster role? It should only need access to objects within the namespace

Reply:
We need to list storage class to check given storage class is exist or not.

Reply:
Only the storageclass permission needs to be in clusterrole then. The other permissions can be moved to Role

Reply:
Changed